### PR TITLE
Fix Ruby 3.0 support with Berkshelf::APIClient::ChefServerConnection

### LINKED ISSUE #1867: Problem using :chef_server as source with Ruby 3.0
## Description

When using `source :chef_server` in a Berksfile using the latest workstation, I get the following backtrace:

```
Resolving cookbook dependencies...
Fetching 'chef-repo' from source at .
Fetching cookbook index from chef_server: https://chef.example.org/organizations/foo...
#<Thread:0x000055ead3744668 /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/installer.rb:21 run> terminated with exception (report_on_exception is true):
/opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/ridley_compat.rb:103:in `initialize': wrong number of arguments (given 1, expected 0) (ArgumentError)
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/api_client/chef_server_connection.rb:11:in `new'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/api_client/chef_server_connection.rb:11:in `initialize'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/api_client.rb:20:in `new'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/api_client.rb:20:in `chef_server'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/source.rb:63:in `api_client'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/source.rb:85:in `build_universe'
        from /opt/cinc-workstation/embedded/lib/ruby/gems/3.0.0/gems/berkshelf-7.2.0/lib/berkshelf/installer.rb:24:in `block (2 levels) in build_universe'
ArgumentError wrong number of arguments (given 1, expected 0)
```

This smells like a Ruby 3.0 issue since this worked fine on the previous workstation release w/ Ruby 2.7.

## Berkshelf Version

7.2.0

## Platform Version

Debian 10

## Berks Output

```

```

### PR DESCRIPTION
**Pull request by ramereth**
Resolves #1867

Also add a simple spec test to ensure this isn't missed again.

Signed-off-by: Lance Albertson <lance@osuosl.org>

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)

### CODE REVIEW

File: lib/berkshelf/api_client/chef_server_connection.rb [hunk 1]
```diff
@@ -5,15 +5,13 @@ module APIClient
     require_relative "errors"
 
     class ChefServerConnection
-      attr_reader :client
-
-      def initialize(*args)
-        @client = Berkshelf::RidleyCompat.new(*args)
-        @url = args[0][:server_url]
+      def initialize(args)
```
Comment (lamont-granquist) [inline lib/berkshelf/api_client/chef_server_connection.rb:8]: should be `def initialize(**args)` here

File: spec/unit/berkshelf/berkshelf/api_client/chef_server_connection_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+describe Berkshelf::APIClient::ChefServerConnection do
+  let(:instance) do
+    described_class.new(
+      {
+        server_url: "https://chef.example.org/organizations/foobar",
+        timeout: 30,
+        open_timeout: 3,
+        ssl: {},
+      }
```
Comment (lamont-granquist) [inline spec/unit/berkshelf/berkshelf/api_client/chef_server_connection_spec.rb:11]: this looks buggy, you don't want the surrounding hash

File: lib/berkshelf/api_client/chef_server_connection.rb (hunk 1 above)
Comment (ramereth) [reply lib/berkshelf/api_client/chef_server_connection.rb]: Fixed

File: spec/unit/berkshelf/berkshelf/api_client/chef_server_connection_spec.rb (hunk 2 above)
Comment (ramereth) [reply spec/unit/berkshelf/berkshelf/api_client/chef_server_connection_spec.rb]: Fixed

**Review by lamont-granquist — APPROVED** [review]

**Pull request merged**